eslint.config.js

```diff
@@ -0,0 +1,259 @@
+import html from 'eslint-plugin-html';
+import markdown from 'eslint-plugin-markdown';
+import globals from 'globals';
+import n from 'eslint-plugin-n';
+import svelte_config from '@sveltejs/eslint-config';
+
+export default [
+	{
+		ignores: [
+			'**/temp/**',
+			'**/dist/**',
+			'**/build/**',
+			'**/.svelte-kit/**',
+			'**/.svelte/**',
+			'packages/playground/big/src/pages/**', // lots of generated files
+			'packages/*/types/index.d.ts',
+			'packages/*/types/index.d.ts.map',
+			'packages/*/CHANGELOG.md'
+		]
+	},
+	...svelte_config, // contains setup for svelte, typescript and unicorn
+	n.configs['flat/recommended-module'],
+	{
+		plugins: {
+			html,
+			markdown
+		},
+
+		languageOptions: {
+			globals: {
+				Atomics: 'readonly',
+				SharedArrayBuffer: 'readonly',
+				$derived: 'readonly',
+				$effect: 'readonly',
+				$props: 'readonly',
+				$state: 'readonly'
+			},
+			ecmaVersion: 2022,
+			sourceType: 'module'
+		},
+
+		rules: {
+			'n/no-unsupported-features/es-builtins': 'error',
+			'n/no-unsupported-features/es-syntax': 'error',
+			'no-console': 'off',
+			'no-debugger': 'error',
+
+			'n/no-missing-import': [
+				'error',
+				{
+					allowModules: ['types', 'estree', 'testUtils', '@sveltejs/vite-plugin-svelte', 'svelte']
+				}
+			],
+
+			'n/no-missing-require': [
+				'error',
+				{
+					allowModules: ['pnpapi']
+				}
+			],
+
+			'n/no-extraneous-import': [
+				'error',
+				{
+					allowModules: ['vite', 'vitest']
+				}
+			],
+
+			'n/no-extraneous-require': [
+				'error',
+				{
+					allowModules: ['vite']
+				}
+			],
+
+			'n/no-deprecated-api': 'off',
```

Comment:
are we using many deprecated APIs? maybe we could just ignore them where we are or whitelist them here if they can't be fixed?